### PR TITLE
feat(upload): Restrict versioning nodes without changes in file_size or md5

### PR DESCRIPTION
**Pull request by kulgan**
Adds an extra check to restrict versioning released nodes that does not include a change in either md5 or file_size

### CODE REVIEW

File: sheepdog/globals.py [hunk 1]
```diff
@@ -224,6 +224,5 @@ def submitted_state():
 
 # Below is the list of node states that are treated as "released"
 RELEASED_NODE_STATES = [
-    'released',
-    'live',
+    'released'
```
Comment (khan08) [inline sheepdog/globals.py]: Probably need to make sure removing live state won't impact current system

Comment (kulgan) [reply sheepdog/globals.py]: Yea, checked the data we have currently, all nodes in state `live` are `File` or `Archive` and we sure don;t want to version those

File: sheepdog/transactions/upload/sub_entities.py [hunk 2]
```diff
@@ -735,3 +691,32 @@ def _create_index(self, **kwargs):
 
     def _version_index(self, **kwargs):
         return self.transaction.indexd.add_version(**kwargs)
+
+    def _should_version_node(self, node):
```
Comment (anmaxvl) [inline sheepdog/transactions/upload/sub_entities.py]: so this prevents versioning in indexd? is the node updated on the graph?

Comment (kulgan) [reply sheepdog/transactions/upload/sub_entities.py]: This prevents calling node_recreate which deletes current node and sets action to `version` is then used to update indexd. All nodes properties always gets updated cos of the way the code is structured. This determines when versioning when versioning should happen

File: dev-requirements.txt [hunk 3]
```diff
@@ -26,3 +26,4 @@ pyasn1-modules==0.0.11
 urllib3==1.22
 -e git+https://github.com/uc-cdis/cirrus.git@0.0.0#egg=cirrus-0.0.0
 -e git+https://git@github.com/uc-cdis/fence.git@1.1.1#egg=fence
+google-resumable-media==0.3.1
```
[the remark this answers is not in the document]
Comment (kulgan) [reply dev-requirements.txt]: anything higher than that only works for python 3

[the remark this answers is not in the document]
Comment (anmaxvl) [reply dev-requirements.txt]: oh, ok. it's the dependency you talked about.

File: sheepdog/transactions/upload/sub_entities.py (hunk 2 above)
Comment (anmaxvl) [reply sheepdog/transactions/upload/sub_entities.py]: 👍

**Comment by khan08** [comment]
Have you tested this in your cluster with real data? I went through the code and it looks fine to me, but seems like there a lot of critical changes.

**Pull request merged**